rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php

```diff
@@ -100,17 +106,35 @@ public function refactor(Node $node): ?Node
                 continue;
             }
 
-            $type = $this->trustedClassMethodPropertyTypeInferer->inferProperty(
-                $node,
+            if (! $classReflection instanceof ClassReflection) {
+                $classReflection = $this->reflectionResolver->resolveClassReflection($node);
+            }
```

Comment:
This should not be needed.

Reply:
it initialized as null before loop, we don't want to get class reflection when class doesn't have property, so early instanceof is used for tweak, then early skip when it got null

Reply:
This rule should be simple as possible. Please remove the `$classReflection = $this->reflectionResolver->resolveClassReflection($node);` and use simple check as before.

Also, wait for review before next merges. I need to review those first

Reply:
sure, see https://github.com/rectorphp/rector-src/pull/6181